benchmark/cmd/dumpkv/main.go

```diff
@@ -93,19 +132,88 @@ func benchmark(cmd *cobra.Command, args []string) {
 	}
 
 	if outputDir == "" {
-		panic("Must provide output dir when generating raw kv data")
+		panic("Must provide output dir")
 	}
 
 	_, isAcceptedBackend := validDBBackends[dbBackend]
 	if !isAcceptedBackend {
 		panic(fmt.Sprintf("Unsupported db backend: %s\n", dbBackend))
 	}
 
-	if modules != "" {
-		exportModules = strings.Split(modules, ",")
+	BenchmarkWrite(rawKVInputDir, outputDir, dbBackend, concurrency, maxRetries, chunkSize)
+}
+
+func benchmarkRead(cmd *cobra.Command, args []string) {
+	if dbBackend == "" {
+		panic("Must provide db backend when benchmarking")
+	}
+
+	if rawKVInputDir == "" {
+		panic("Must provide raw kv input dir when benchmarking")
+	}
+
+	if outputDir == "" {
+		panic("Must provide output dir")
+	}
+
+	_, isAcceptedBackend := validDBBackends[dbBackend]
+	if !isAcceptedBackend {
+		panic(fmt.Sprintf("Unsupported db backend: %s\n", dbBackend))
+	}
+
+	BenchmarkRead(rawKVInputDir, outputDir, dbBackend, concurrency, maxRetries, chunkSize)
+}
+
+func benchmarkForwardIteration(cmd *cobra.Command, args []string) {
+	if dbBackend == "" {
+		panic("Must provide db backend when benchmarking")
+	}
+
+	if outputDir == "" {
+		panic("Must provide output dir")
+	}
+
+	_, isAcceptedBackend := validDBBackends[dbBackend]
+	if !isAcceptedBackend {
+		panic(fmt.Sprintf("Unsupported db backend: %s\n", dbBackend))
+	}
+
+	if prefixes != "" {
+		forwardIterationPrefixes = strings.Split(prefixes, ",")
+	} else {
+		// Use module prefixes as default
+		for _, module := range modules {
+			modulePrefix := fmt.Sprintf("s/k:%s/", module)
+			forwardIterationPrefixes = append(forwardIterationPrefixes, modulePrefix)
+		}
+	}
+
+	BenchmarkDBIteration(forwardIterationPrefixes, outputDir, dbBackend, concurrency)
+}
+
+func benchmarkReverseIteration(cmd *cobra.Command, args []string) {
+	if dbBackend == "" {
+		panic("Must provide db backend when benchmarking")
+	}
+
+	if outputDir == "" {
+		panic("Must provide output dir")
 	}
 
-	BenchmarkWrite(rawKVInputDir, exportModules, outputDir, dbBackend, concurrency, maxRetries, chunkSize)
+	_, isAcceptedBackend := validDBBackends[dbBackend]
+	if !isAcceptedBackend {
+		panic(fmt.Sprintf("Unsupported db backend: %s\n", dbBackend))
+	}
+
+	// Specify prefixes here. Can forward iterate and retrieve keys
+	// NOTE: Will add reference keys + add in random prefix iteration
```

Comment:
As noted, can prespecify reverse prefixes after the forward iteration. Also will be adding in random key reverse iteration